File: app/src/main/java/net/devdome/paperplayer/adapter/PlaylistAdapter.java
package net.devdome.paperplayer.adapter;


import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import net.devdome.paperplayer.R;
import net.devdome.paperplayer.playback.PlaylistManager;

import java.util.ArrayList;

public class PlaylistAdapter extends RecyclerView.Adapter<PlaylistAdapter.ViewHolder> {

    private ArrayList<PlaylistManager.PlaylistItem> playlist;

    public PlaylistAdapter() {
        PlaylistManager playlistManager = PlaylistManager.getInstance();
        playlist = playlistManager.getCurrentPlaylist();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_playlist_item, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.title.setText(playlist.get(position).getSong().getName());
        holder.artist.setText(playlist.get(position).getSong().getArtist());

    }

    @Override
    public int getItemCount() {
        return playlist.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        View itemView;
        TextView title, artist;

        public ViewHolder(View itemView) {
            super(itemView);
            this.itemView = itemView;
            title = (TextView) itemView.findViewById(R.id.song_item_title);
            artist = (TextView) itemView.findViewById(R.id.song_item_artist);
        }
    }
}

File: app/src/main/java/xyz/michaelobi/paperplayer/presentation/player/playlist/PlaylistAdapter.kt
/*
 * MIT License
 *
 * Copyright (c) 2017 MIchael Obi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package xyz.michaelobi.paperplayer.presentation.player.playlist

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.widget.TextView
import xyz.michaelobi.paperplayer.R
import xyz.michaelobi.paperplayer.playback.queue.QueueItem


/**
 * PaperPlayer
 * Michael Obi
 * 22 05 2017 11:19 PM
 */

class PlaylistAdapter : androidx.recyclerview.widget.RecyclerView.Adapter<PlaylistAdapter.ViewHolder>() {
    private var queue: List<QueueItem> = emptyList()

    fun setQueueItems(queue: List<QueueItem>) {
        this.queue = queue
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: android.view.ViewGroup, viewType: Int): PlaylistAdapter.ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.card_playlist_item, parent, false)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: PlaylistAdapter.ViewHolder, position: Int) {
        holder.title.text = queue[position].song.title
        holder.artist.text = queue[position].song.artist
    }

    override fun getItemCount(): Int = queue.size

    class ViewHolder(itemView: android.view.View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {
        var title = itemView.findViewById(R.id.song_item_title) as TextView
        var artist = itemView.findViewById(R.id.song_item_artist) as TextView
    }
}
